scylla-cdc-replicator/src/replicator_consumer.rs

```diff
@@ -27,6 +27,9 @@ pub(crate) struct ReplicatorConsumer {
 
     // Strings for queries created dynamically:
     keys_cond: String,
+
+    // Stores data for left side range delete while waiting for its right counterpart.
+    left_range_details: (Option<usize>, bool, Option<CqlValue>),
```

Comment:
This data is not enough to reconstruct a range delete. Take into account that queries like the following also exist:

```cql
DELETE FROM ks.t WHERE (ck1, ck2) > (1, 2);
DELETE FROM ks.t WHERE (ck1, ck2) > (1, 2) AND (ck1, ck2) < (9, 10);
DELETE FROM ks.t WHERE (ck1, ck2, ck3, ck4) > (1, 2, 3, 4) AND (ck1, ck2) < (9, 10);
```

You will have to keep all `CqlValue`s from the previous row (or at least up to the first `null`).

Reply:
Moreover, please convert the tuple either into separate fields with clear names, or a struct - also with clear field names.